### DISCUSSION TITLE
Capstone ML - Where Were You -Error

### DISCUSSION THREAD

**Discussion by moogli90** [teamalgoritma/community#347] [Q&A] [answered]
Mohon bantuan kakak, kenapa muncul error seperti ini?
![Capstone ml1](https://user-images.githubusercontent.com/107478291/192138324-4c6acf04-60b6-488e-8294-b537041d1773.png)

**Comment by fafilia** [comment]
Halo @moogli90 ,
Pertanyaan Anda sepertinya menyerupai diskusi ini: https://stackoverflow.com/questions/59169441/importerror-could-not-import-pil-image/. Apakah bisa dicoba untuk melakukan install pillow via conda. Caranya dengan run code berikut di terminal/anaconda prompt pada environment Bapak: `conda install pillow`

Demikian yang dapat saya sampaikan, semoga bermanfaat.

**Reply by moogli90** [reply]
sudah sy coba install pillow via terminal maupun cona promt tapi masih belum bisa jalan
![image](https://user-images.githubusercontent.com/107478291/192228312-86a3b265-1656-4753-ac37-bf4c079682ee.png)

**Comment by vetricko** [comment]
Memastikan terlebih dahulu dalam environment terdapat pillow di anaconda prompt : 
> conda activate -*nama environtment*
> conda list

<img width="461" alt="image" src="https://user-images.githubusercontent.com/65656955/192231373-a3780515-67d8-4673-babb-155f96dfeeaf.png">

Setelah itu di activate environmentnya di rstudio, kalau saya pakai library reticulate.
library(reticulate)
use_condaenv(*nama environment*)
<img width="196" alt="image" src="https://user-images.githubusercontent.com/65656955/192232567-461eefc8-7273-49cd-bf0f-ddd738e3c2d9.png">

Semoga membantu🙏

Note tambahan: jika di environment sudah ada pillow, dan masih error. Bisa di coba tutup dan buka ulang r studionya.

**Reply by moogli90** [reply]
Sudah sy buka tutup dan buka ulang rstudio, dan activate environmentnya di rstudio namun tetap tidak bisa kak? 
![image](https://user-images.githubusercontent.com/107478291/192247101-7de8c5c6-84e4-44d9-95ec-a9309507bf8f.png)

**Reply by vetricko** [reply]
sy pernah dapat masalah ini sebelumnya, tetapi setelah sy restart laptop karena masalah dengan google chrome sudah bisa dijalankan. Apakah boleh dicoba di restart terlebih dahulu? 

Dari website https://kegui.medium.com/stopiteration-could-not-import-pil-image-the-use-of-array-to-img-requires-pil-cea3802ba27d mengingatkan untuk di restart anacondanya.

**Comment by fafilia** [comment, marked as answer]
Jika dilihat dari list library pada environment yang dibuat sudah ada library Pillow. Mungkin bisa dibantu cek pak, apakah di settingan global optionnya juga sudah disetting untuk environment r-tensorflow?

Silahkan mengikuti cara berikut untuk setting di global options RStudio. Kilk Tools-> Global Options... -> Python -> Pilih/cari environment environment

![image](https://user-images.githubusercontent.com/14083457/192253910-75e61186-84f3-4aa0-9f82-d6c06a20ff23.png)

**Reply by moogli90** [reply]
dari awal sudah sy seeting r-tensorflow kak, sekarang errornya berbeda?
![image](https://user-images.githubusercontent.com/107478291/192280794-08f76336-f003-4bcd-a002-e9ee1b4f3914.png)

**Reply by fafilia** [reply]
Dari error baru yang muncul sepertinya dibutuhkan library scipy untuk bisa menjalankan proses tersebut Pak @moogli90 . Silahkan dicoba untuk diinstal menggunakan perintah pip pada Anaconda Prompt. Jangan lupa untuk restart kembali Anaconda dan R/Rstudionya.
